### PR TITLE
Autodetect clients based on their master public key fingerprint

### PR DESCRIPTION
**Pull request by achow101**
This PR displays the master public key fingerprint in the enumerate output and allows for the user to specify the fingerprint instead of the device type and path so that the device can be automatically detected.

### CODE REVIEW

**Comment by instagibbs** [comment]
Fails with my nano s.

```
  File "./hwi.py", line 11, in <module>
    result = process_commands(sys.argv[1:])
  File "/home/greg/bitcoin-dev/HWI/hwilib/commands.py", line 217, in process_commands
    return args.func()
  File "/home/greg/bitcoin-dev/HWI/hwilib/commands.py", line 85, in enumerate
    client = get_client(d_data['type'], d_data['path'])
  File "/home/greg/bitcoin-dev/HWI/hwilib/commands.py", line 46, in get_client
    client = ledgeri.LedgerClient(device=device)
  File "/home/greg/bitcoin-dev/HWI/hwilib/ledgeri.py", line 22, in __init__
    self.dongle = getDongle(True)
  File "/home/greg/bitcoin-dev/btchip-python/btchip/btchipComm.py", line 230, in getDongle
    dev.open_path(hidDevicePath)
  File "hid.pyx", line 72, in hid.device.open_path
OSError: open failed
```

I'll debug later.

**Comment by achow101** [comment]
@instagibbs Apparently I had some changes that I forgot to push. I don't know if they have any effect on your problem though, but can you try again?

File: hwilib/commands.py [hunk 1]
```diff
@@ -24,33 +24,84 @@
...
+            d_data['type'] = 'coldcard'
+        else:
+            continue
+        d_data['path'] = d['path'].decode("utf-8")
+        d_data['serial_number'] = d['serial_number']
+
+        client = get_client(d_data['type'], d_data['path'])
+        master_xpub = client.get_pubkey_at_path('m/0h')['xpub']
+        d_data['fingerprint'] = get_xpub_fingerprint_hex(master_xpub)
+        client.close()
```
Comment (instagibbs) [inline hwilib/commands.py]: yes this is what fixed it for me too. Will test.

**Comment by instagibbs** [comment]
tACK ledger https://github.com/achow101/HWI/pull/29/commits/c21c040ddc51cee7b1a937bba6916c39654f95f7

**Pull request merged**